Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/GameAdapter.java

```diff
@@ -118,83 +87,33 @@ public void onBindViewHolder(GameViewHolder holder, int position)
 	@Override
 	public int getItemCount()
 	{
-		if (mDatasetValid && mCursor != null)
-		{
-			return mCursor.getCount();
-		}
-		Log.error("[GameAdapter] Dataset is not valid.");
-		return 0;
-	}
-
-	/**
-	 * Return the contents of the _id column for a given row.
-	 *
-	 * @param position The row for which Android wants an ID.
-	 * @return A valid ID from the database, or 0 if not available.
-	 */
-	@Override
-	public long getItemId(int position)
-	{
-		if (mDatasetValid && mCursor != null)
-		{
-			if (mCursor.moveToPosition(position))
-			{
-				return mCursor.getLong(GameDatabase.COLUMN_DB_ID);
-			}
-		}
-
-		Log.error("[GameAdapter] Dataset is not valid.");
-		return 0;
+		return mGameFiles.size();
 	}
 
 	/**
 	 * Tell Android whether or not each item in the dataset has a stable identifier.
-	 * Which it does, because it's a database, so always tell Android 'true'.
 	 *
 	 * @param hasStableIds ignored.
 	 */
 	@Override
 	public void setHasStableIds(boolean hasStableIds)
 	{
-		super.setHasStableIds(true);
+		super.setHasStableIds(false);
 	}
 
 	/**
-	 * When a load is finished, call this to replace the existing data with the newly-loaded
-	 * data.
-	 *
-	 * @param cursor The newly-loaded Cursor.
+	 * When a load is finished, call this to replace the existing data
+	 * with the newly-loaded data.
 	 */
-	public void swapCursor(Cursor cursor)
+	public void swapDataSet(List<GameFile> gameFiles)
 	{
 		// Sanity check.
-		if (cursor == mCursor)
+		if (gameFiles == mGameFiles)
```

Comment:
Wouldn't ``==`` only compare the pointer of the list not its contents? So two identical lists would return false here if they're not stored at the same location (as the same object). 

I guess it's intentional here but I thought it might be worth pointing out

Reply:
Yes. I don't think there's any point to this check, but it doesn't hurt either... Would you prefer that I get rid of it?

Reply:
If it doesn't do anything useful, better get rid off it. (Although if you're too lazy to, I don't care all that much :P)